app/src/main/java/im/koala/bcsd/ui/login/LoginActivity.kt

```diff
@@ -0,0 +1,496 @@
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
+import im.koala.bcsd.R
+import im.koala.bcsd.ui.findid.FindIdActivity
+import im.koala.bcsd.ui.findpassword.FindPasswordActivity
+import im.koala.bcsd.ui.join.JoinActivity
+import im.koala.bcsd.ui.theme.Black
+import im.koala.bcsd.ui.theme.Gray
+import im.koala.bcsd.ui.theme.GrayBorder
+import im.koala.bcsd.ui.theme.GrayDisabled
+import im.koala.bcsd.ui.theme.GrayNormal
+import im.koala.bcsd.ui.theme.Green
+import im.koala.bcsd.ui.theme.KoalaTheme
+import im.koala.bcsd.ui.theme.White
+import im.koala.bcsd.ui.theme.Yellow
+import im.koala.bcsd.ui.theme.Yellow2
+
+class LoginActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            KoalaTheme {
+                // A surface container using the 'background' color from the theme
+                Surface(color = MaterialTheme.colors.background) {
+                    constraintLayoutContent()
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun constraintLayoutContent() {
+    val context = LocalContext.current
+    var loginTabState = remember { mutableStateOf(true) }
+    ConstraintLayout(modifier = Modifier.fillMaxSize()) {
+        val (logoImageView, loginRowLayout, normalLoginConstraintLyaout, snsLoginConstraintLyaout) = createRefs()
+
+        drawImageView(
+            modifier = Modifier.constrainAs(logoImageView) {
+                top.linkTo(parent.top, margin = 48.dp)
+                start.linkTo(parent.start)
+                end.linkTo(parent.end)
+            },
+            drawableId = R.drawable.ic_koala_logo
+        )
+        Row(
+            modifier = Modifier.constrainAs(loginRowLayout) {
+                top.linkTo(logoImageView.bottom, margin = 40.dp)
+                start.linkTo(parent.start, margin = 16.dp)
+                end.linkTo(parent.end, margin = 16.dp)
+                width = Dimension.fillToConstraints
+            }.size(0.dp, 71.dp)
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .weight(1f)
+            ) {
+                TextButton(
+                    onClick = { loginTabState.value = true },
+                    colors = ButtonDefaults.buttonColors(
+                        backgroundColor = Color.Transparent,
+                    ),
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .height(70.dp)
+                ) {
+                    Text(
+                        text = stringResource(id = R.string.normal_login),
+                        color = Black,
+                        style = MaterialTheme.typography.h2,
+                    )
+                }
+                if (loginTabState.value) {
```

Comment:
delegation을 사용하면 깔끔하게 짤 수 있습니다.

```suggestion
                if (loginTabState) {
```

Reply:
loginTabState 가 어떤 역할을 하는지 모르겠습니다.

Reply:
현재 화면이 일반로그인인지 sns로그인 화면인지 구분하는 변수입니다.
변수의 의미를 좀더 정확히 하기위해 isNormalLoginState로 변경하겠습니다.